Models/User.go

```diff
@@ -1,52 +1,63 @@
-package Models
+package model
 
-import(
+import (
+	"first-api/config"
 	"fmt"
-	"first-api/Config"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/go-ozzo/ozzo-validation/v4/is"
 )
 
-// get all users Fetch all user data
- func GetAllUsers(user *[]User) (err error){
-	if err = Config.DB.Find(user).Error; err !=nil{
+type UserStore struct{}
+
+func (u *UserStore) CreateUser(user *User) error {
+	if err := config.DB.Create(user).Error; err != nil {
 		return err
 	}
 	return nil
- }
+}
 
- // CreateUser
+func (us *UserStore) Validate(u User) error {
+	return validation.ValidateStruct(&u, validation.Field(&u.Name, validation.Required, validation.Length(5, 20)),
+		validation.Field(&u.Email, validation.Required, is.Email),
+		validation.Field(&u.Phone, validation.Required, validation.Length(10, 10)),
+		validation.Field(&u.Address, validation.Required, validation.Length(10, 50)),
+	)
```

Comment:
Validation should not be part of your models or store rather it should have been part of your controller.